File: content/snippets/typescript/language_basics/strings.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";

function main(): void {
  const first = "hello";
  const second = "world";
  const joined = `${first} ${second}`;

  const part = joined.slice(0, 5);
  const replaced = joined.replace("hello", "hi");
  const words = joined.split(/\s+/);

  console.log("joined:", joined);
  console.log("part:", part);
  console.log("replaced:", replaced);
  console.log("tokens:", words);
}

main();
setImmediate(() => {});

File: content/snippets/go/language_basics/strings.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	first := "hello"
	second := "world"
	joined := first + " " + second

	part := joined[:5]
	replaced := strings.Replace(joined, "hello", "hi", 1)
	words := strings.Fields(joined)

	fmt.Println("joined:", joined)
	fmt.Println("part:", part)
	fmt.Println("replaced:", replaced)
	fmt.Println("tokens:", words)
}
